photon-test/src/main/scala/com/linkedin/photon/ml/test/TestTemplateWithTmpDir.scala

```diff
@@ -35,7 +35,7 @@ trait TestTemplateWithTmpDir {
 }
 
 private object TestTemplateWithTmpDir {
-  private val _tmpDirThreadLocal: ThreadLocal[String] = new ThreadLocal[String] {
+  private def _tmpDirThreadLocal: ThreadLocal[String] = new ThreadLocal[String] {
```

Comment:
Just curious: what was the bug? This is a pretty big change to the semantics of ```getTmpDir```. Previously, it defined a temporary base dir that would be constant for all tests running in a particular test class and thread. With this change, a new directory is created for each call to ```getTmpDir```. Also, this doesn't really need to be a ```ThreadLocal``` anymore, since it's not shared between the threads.